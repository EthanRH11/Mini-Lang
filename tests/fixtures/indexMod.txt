begin:

    elements<int> arr;

    arr = range(0..20);
    out_to_console(arr);
    ...
    out_to_console("***************************************");
    ...
    out_to_console("Index prior to operation: " + arr@(5));
    ...
    arr.(5 + 20);
    out_to_console("Index after addition operation: " + arr@(5));
    ...
    out_to_console("Index prior to operation: " + arr@(8));
    ...
    arr.(8 * 2);
    out_to_console("Index after multiplication operation: " + arr@(8));
    ...
    out_to_console("Index prior to operation: " + arr@(11));
    ...
    arr.(11 - 4);
    out_to_console("Index after subtraction operation: " +  arr@(11));
    ...
    out_to_console("Index prior to operation: " + arr@(18));
    ...
    arr.(18 / 2);
    out_to_console("Index after division operation: " + arr@(18));
    ...
    out_to_console("Index prior to operation: " + arr@(20));
    ...
    arr.(20 % 2);   
    out_to_console("Index after modulus operation: " + arr@(20));


end