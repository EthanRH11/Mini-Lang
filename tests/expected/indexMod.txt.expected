[0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20]
***************************************
Index prior to operation: 5
Index after addition operation: 25
Index prior to operation: 8
Index after multiplication operation: 16
Index prior to operation: 11
Index after subtraction operation: 7
Index prior to operation: 18
Index after division operation: 9
Index prior to operation: 20
Index after modulus operation: 0